src/vmm/src/device_manager/mmio.rs

```diff
@@ -94,12 +95,6 @@ pub struct MMIODeviceManager {
 impl MMIODeviceManager {
     /// Create a new DeviceManager handling mmio devices (virtio net, block).
     pub fn new(mmio_base: &mut u64, irq_interval: (u32, u32)) -> MMIODeviceManager {
-        if cfg!(target_arch = "aarch64") {
```

Comment:
Since the `mmio_base` is not changed anymore, the input parameter shouldn't be mut.

This change needs to be propagated in `builder.rs` and we can then removed the `allow(unused_mut)`: https://github.com/firecracker-microvm/firecracker/blob/e630dfc62322cf7eaf17aa3b65d6d5a82f535bc0/src/vmm/src/builder.rs#L221

Reply:
Done,  but this also caught an unused `mut` on the `mmio_device_manager`.  I removed it,  but maybe a sanity check would be sensible.  Also removed what I think in an unused cast here.

Reply:
Thanks!